more_itertools/more.py

```diff
@@ -1248,19 +1245,21 @@ def numeric_range(*args):
         >>> list(numeric_range(start, stop, step))
         [Fraction(1, 2), Fraction(1, 1), Fraction(3, 2), Fraction(2, 1)]
 
-    Negative steps are supported:
+    If *step* is zero,``ValueError`` is rasied. Negative steps are supported:
 
         >>> list(numeric_range(3, -1, -1.0))
         [3.0, 2.0, 1.0, 0.0]
 
+    The yielded items will have the same type as the first argument.
```

Comment:
except when the addition of `start` and a multiple of `step` "coerces" to another type. :)